packages/device_info/lib/device_info.dart

```diff
@@ -0,0 +1,231 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
```

Comment:
Would the user have to always cast to one of the concrete classes to extract anything from the info object? I think if we want a least common denominator info object, we should add it as a third-option, then you can do:

```dart
// No type casting
plugin.iosInfo.identifierForVendor;
plugin.androidInfo.version.previewSdkInt;
plugin.info.name;
```